tests/dotnet/UnitTests/ResourcePrefixTest.cs

```diff
@@ -0,0 +1,138 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+
+namespace Xamarin.Tests {
+	[TestFixture]
+	public class ResourcePrefixTest : TestBaseClass {
+		[Test]
+		public void ResourcePrefix_DefaultValues ()
+		{
+			// Arrange
+			var platforms = Configuration.GetAllPlatforms ();
+
+			foreach (var platform in platforms) {
+				// Act & Assert
+				var defaultValue = GetResourcePrefix (platform);
+				Assert.AreEqual ("Resources", defaultValue, $"Default value for {platform} should be 'Resources'");
+			}
+		}
+
+		[Test]
+		public void ResourcePrefix_AppBundleResourcePrefix ()
+		{
+			// Arrange
+			var platforms = Configuration.GetAllPlatforms ();
+			var customPrefix = "CustomResources";
+
+			foreach (var platform in platforms) {
+				// Act
+				var value = GetResourcePrefix (platform, ("AppBundleResourcePrefix", customPrefix));
+
+				// Assert
+				Assert.AreEqual (customPrefix, value, $"{platform}: AppBundleResourcePrefix should be used");
+			}
+		}
+
+		[Test]
+		public void ResourcePrefix_PlatformSpecific ()
+		{
+			// Arrange
+			var customPrefix = "CustomResources";
+
+			// Act & Assert
+
+			// iOS and tvOS use IPhoneResourcePrefix
+			var iOSValue = GetResourcePrefix ("ios", ("IPhoneResourcePrefix", customPrefix));
+			Assert.AreEqual (customPrefix, iOSValue, "iOS should use IPhoneResourcePrefix");
```

Comment:
Use `Configuration.include_ios` to check whether iOS is currently enabled. Do the equivalent change for the other platforms as well.